From 2f160c435f36cf839104ac56eae878d2608bbdf3 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 16 Jun 2020 14:28:00 -0500
Subject: [PATCH 39/39] arm64: dts: beacon-imx8mm-hdmi: Fix HDMI regulator

The HDMI regulator has the wrong polarity, so it doesn't properly
power on.

This patch changes the polarity of the enable signal, and references
the regulator from the HDMI bridge so it can be powered on and off
instead of keeping it on all the time.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
index e0e795b36c7c..270412e6a7bb 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
@@ -24,9 +24,9 @@
 		regulator-name = "hdmi_pwr_en";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio2 11 GPIO_ACTIVE_LOW>;
-		startup-delay-us = <500000>;
-		regulator-always-on;
+		gpio = <&gpio2 11 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		startup-delay-us = <70000>;
 	};
 };
 
@@ -34,6 +34,7 @@
 	adv_bridge: adv7535@3d {
 		compatible = "adi,adv7533";
 		reg = <0x3d>;
+		dvdd-supply = <&reg_hdmi>;
 		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
 		pinctrl-names = "default";
-- 
2.17.1

